File: app/src/main/java/com/popalay/cardme/presentation/screens/splash/SplashActivity.java
package com.popalay.cardme.presentation.screens.splash;

import com.arellomobile.mvp.presenter.InjectPresenter;
import com.popalay.cardme.presentation.base.BaseActivity;
import com.popalay.cardme.presentation.screens.home.HomeActivity;

public class SplashActivity extends BaseActivity implements SplashView {

    @InjectPresenter SplashPresenter presenter;

    @Override public void openHome() {
        startActivity(HomeActivity.Companion.getIntent(this));
        finish();
    }
}

File: app/src/main/java/com/popalay/cardme/presentation/screens/splash/SplashActivity.kt
package com.popalay.cardme.presentation.screens.splash

import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import com.popalay.cardme.presentation.base.BaseActivity
import javax.inject.Inject

class SplashActivity : BaseActivity() {

    @Inject lateinit var factory: ViewModelProvider.Factory

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ViewModelProviders.of(this, factory).get(SplashViewModel::class.java)
    }
}
